internal/mode/static/telemetry/platform.go

```diff
@@ -0,0 +1,101 @@
+package telemetry
+
+import (
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+type k8sState struct {
+	node       v1.Node
+	namespaces v1.NamespaceList
+}
+
+type platformExtractor func(k8sState) (string, bool)
+
+func buildProviderIDExtractor(id string, platform string) platformExtractor {
+	return func(state k8sState) (string, bool) {
+		if strings.HasPrefix(state.node.Spec.ProviderID, id) {
+			return platform, true
+		}
+		return "", false
+	}
+}
+
+const (
+	gkeIdentifier       = "gce"
+	awsIdentifier       = "aws"
+	azureIdentifier     = "azure"
+	kindIdentifier      = "kind"
+	k3sIdentifier       = "k3s"
+	openshiftIdentifier = "node.openshift.io/os_id"
+	rancherIdentifier   = "cattle-system"
+
+	platformGKE       = "gke"
+	platformAWS       = "eks"
+	platformAzure     = "aks"
+	platformKind      = "kind"
+	platformK3S       = "k3s"
+	platformOpenShift = "openshift"
+	platformRancher   = "rancher"
+	platformOther     = "other"
+)
+
+var platformExtractors = []platformExtractor{
+	openShiftExtractor,
+	rancherExtractor,
+	// ID provider extractors must run after the rest
+	buildProviderIDExtractor(gkeIdentifier, platformGKE),
+	buildProviderIDExtractor(awsIdentifier, platformAWS),
+	buildProviderIDExtractor(azureIdentifier, platformAzure),
+	buildProviderIDExtractor(kindIdentifier, platformKind),
+	buildProviderIDExtractor(k3sIdentifier, platformK3S),
+}
+
+func getPlatform(node v1.Node, namespaces v1.NamespaceList) string {
+	state := k8sState{
+		node:       node,
+		namespaces: namespaces,
+	}
+
+	for _, extractor := range platformExtractors {
+		if platform, ok := extractor(state); ok {
+			return platform
+		}
+	}
+
+	return unknownProviderIDExtractor(state)
+}
+
+func openShiftExtractor(state k8sState) (string, bool) {
+	// openshift platform won't show up in node's ProviderID
+	if value, ok := state.node.Labels[openshiftIdentifier]; ok && value != "" {
```

Comment:
I think you can simplify this as well. If the label doesn't exist, `value` will be empty string.

Reply:
I was going off of this other solution I found online [here](https://github.com/cisco-open/cluster-registry-controller/blob/3086c2cb8cd559891bc19d746b2f1f89721a77ed/pkg/clustermeta/distributions.go#L256) , where I think they make sure to check that if the label does exist, then `value` should also not be an empty string, if so then we can mark it as openshift. 

I think it would be an edge case for the label to exist and yet the value to be an empty string, but I don't see the harm in having the check in there. Do you think its uncessary?

Reply:
There's no harm, I just think it's redundant. If the value is the empty string, then we can't mark it as openshift, right?  And we know that if the label doesn't exist, then the value will also be empty string.

So, we should be able to simplify to:


```suggestion
	if state.node.Labels[openshiftIdentifier] != "" {
```